addons/survey/static/src/views/widgets/integer_update_flag_field/integer_update_flag_fields.js

```diff
@@ -21,17 +22,18 @@ export class IntegerUpdateFlagField extends IntegerField {
     static props= {
         ...IntegerField.props,
         flagFieldName: { type: String },
-        referenceValue: { type: Number },
     }
     /**
      * @override
      */
     setup() {
         super.setup(...arguments);
+        this.referenceValue = this.props.record?.evalContext?.parent?.session_speed_rating_time_limit;
+        this.orm = useService("orm");
         const inputRef = useRef("numpadDecimal");
         const onChange = async () => {
             await this.props.record._update({
-                [this.props.flagFieldName]: parseInt(this.formattedValue) !== this.props.referenceValue}
+                [this.props.flagFieldName]: parseInt(this.formattedValue) !== this.referenceValue}
```

Comment:
Is it possible that the question get linked to another survey? In that case, we should maybe observe changes made on `this.props.record.data.survey_id` and reset the reference value accordingly? Currently, the reference value becomes static and is only loaded when starting the widget.

(same question for the other widget)

Reply:
Cool. We hence do not need to add code to make it dynamic :smiley: